src/custom/api/gnosisProtocol/api.ts

```diff
@@ -274,12 +277,23 @@ const UNHANDLED_ORDER_ERROR: ApiErrorObject = {
   description: ApiErrorCodeDetails.UNHANDLED_CREATE_ERROR,
 }
 
-async function _handleQuoteResponse<T = any, P extends QuoteQuery = QuoteQuery>(
+async function _handleQuoteResponse<T = any, P extends MinimumSentryQuoteData = MinimumSentryQuoteData>(
   response: Response,
   params?: P
 ): Promise<T> {
   if (!response.ok) {
-    const errorObj: ApiErrorObject = await response.json()
+    const errorObj: ApiErrorObject = await response.json().catch((error) => {
+      const sentryError = new Error()
+      Object.assign(sentryError, error, {
+        message: `Potential backend error detected - status code: ${response.status}`,
+        name: 'HandleQuoteResponseJsonParse',
+      })
+      // report to sentry
+      Sentry.captureException(error, {
+        tags: { errorType: 'handleQuoteResponse', backendErrorCode: response.status },
+        contexts: { params: { ...params } },
+      })
+    })
```

Comment:
I'm a bit concerned with adding now Sentry dependencies everywhere in the code. Can we at least abstract it in a library?

Reply:
done